aten/src/ATen/native/Distance.cpp

```diff
@@ -55,38 +67,73 @@ Tensor cdist(const Tensor& x1, const Tensor& x2, const double p) {
 Tensor _pdist_forward(const Tensor& self, const double p) {
   AT_CHECK(self.is_contiguous(), "_pdist_forward requires contiguous input");
   auto device = self.type().device_type();
-  AT_CHECK(device == kCPU || device == kCUDA, "_pdist_forward only supports CPU and CUDA devices, got: ", device);
-  Tensor result = at::empty({0}, self.options());
-  if (self.size(0) <= 1) {
-    result.resize_({0});
+  AT_CHECK(
+      device == kCPU || device == kCUDA,
+      "_pdist_forward only supports CPU and CUDA devices, got: ",
+      device);
+
+  const auto batches = self.sizes().slice(0, self.dim() - 2);
+  int64_t b = at::tensor(batches).prod().item<int64_t>();
```

Comment:
Is it necessary to use tensor operations for this? If it only operates on sizes() you could also use a simple for loop to accumulate this into `b`.

Reply:
I'd also stay away from those, for example, I've found std::accumulate to have bad performance characteristics.